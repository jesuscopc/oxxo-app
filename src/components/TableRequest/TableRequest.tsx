import React from 'react';
import { Link } from 'react-router-dom';
type TItem = {
  original_price: string;
  product_detail_url: string;
  product_id: string;
  product_main_image_url: string;
  product_title: string;
};

type TData = {
  data: TItem[];
};

const TableRequest = ({ data }: TData) => {
  return (
    <ul>
      {data.map(
        ({
          product_main_image_url,
          product_title,
          original_price,
          product_detail_url,
          product_id
        }) => (
          <li className="flex flex-col p-5 mt-5 rounded-xl md:flex-row bg-white" key={product_id}>
            <img
              className="rounded-lg mx-auto md:m-0"
              width="200px"
              height="200px"
              src={product_main_image_url}
            />
            <div className="flex flex-col p-5">
              <p className="font-bold pb-4">{product_title}</p>
              <span className="text-4xl pb-4">{original_price}</span>
              <Link
                className="text-blue-500 font-extrabold underline"
                tabIndex={0}
                to={product_detail_url}
              >
                Detalle producto
              </Link>
            </div>
          </li>
        )
      )}
    </ul>
  );
};

export default TableRequest;
