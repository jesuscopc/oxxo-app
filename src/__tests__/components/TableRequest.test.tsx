import * as React from 'react';
import { shallow } from 'enzyme';
import TableRequest from 'components/TableRequest/TableRequest';

describe('SUit test TableRequest', () => {
  test('should be render correctly', () => {
    const wrapper = shallow(<TableRequest data={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render a list with items', () => {
    const items = [
      {
        original_price: '3999.00',
        product_detail_url: 'Motorola G',
        product_id: 'JSDSH878',
        product_main_image_url: 'http://motorola.jpeg',
        product_title: 'Motorola G'
      }
    ];
    const wrapper = shallow(<TableRequest data={items} />);
    expect(wrapper.find('span').text()).toBe(items[0].original_price);
  });
});
