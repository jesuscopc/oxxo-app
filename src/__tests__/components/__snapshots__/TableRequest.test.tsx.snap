// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SUit test TableRequest should be render correctly 1`] = `<ul />`;
